### PR TITLE
games-arcade/circuslinux: EAPI8, ebuild improvements

### PR DESCRIPTION
**Pull request by mm1ke**
Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Michael Mair-Keimberger <mmk@levelnine.at>

Another EAPI8 bump for `games-arcade/circuslinux`, this fixes:
* calling gcc directly
* fixes SRC_URI
* removes hardcoded `-Wall -O2'
* fixes installation of doc's

### CODE REVIEW

File: games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
-	sed -i \
-		-e 's/\$(pkgdatadir)/$(DESTDIR)&/' \
-		data/Makefile.in \
-		|| die
+	tc-export CC
+	sed -i -e 's/-Wall -O2//' \
+		Makefile.in || die
+	sed -i -e 's/\$(pkgdatadir)/$(DESTDIR)&/' \
+		data/Makefile.in || die
 }
```
Comment (ionenwks) [inline games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild:30]: This seem to have working joystick support but disables it by default, I don't see much reason to add a IUSE for this (joystick on libsdl is cheap and default) but also no reason to keep it disabled.
```suggestion

src_configure() {
	tc-export CC
	econf --enable-joystick
}

```
If defining src_configure, makes sense to move tc-export here too.

File: games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild [hunk 2]
```diff
@@ -1,37 +1,36 @@
...
 HOMEPAGE="http://www.newbreedsoftware.com/circus-linux/"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~alpha ~amd64 ~x86"
-IUSE=""
 
-DEPEND="media-libs/libsdl
+DEPEND="
+	media-libs/libsdl[sound,video]
```
Comment (ionenwks) [inline games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild:17]: Need revbump for this change and, if adding --enable-joystick, obviously joystick :)

File: games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild [hunk 3]
```diff
@@ -1,37 +1,36 @@
...
-	sed -i \
-		-e "/^install-data-am/d" \
-		Makefile.in \
-		|| die
-	sed -i \
-		-e 's/\$(pkgdatadir)/$(DESTDIR)&/' \
-		data/Makefile.in \
-		|| die
+	tc-export CC
+	sed -i -e 's/-Wall -O2//' \
```
[the remark this answers is not in the document]
Comment (mm1ke) [reply games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild]: I've changed it now to only remove the `-O2`.

File: games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild (hunk 1 above)
Comment (mm1ke) [reply games-arcade/circuslinux/circuslinux-1.0.3-r1.ebuild]: Nice catch, didn't noticed it. Added your suggestion and changed the deps and also made a rev-bump :)

**Review by ionenwks — APPROVED** [review]
Along with the other PRs, thanks for all :)

**Pull request closed without merging**